kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/extensions/KroxyliciousExtension.java

```diff
@@ -49,17 +50,22 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
         Parameter parameter = parameterContext.getParameter();
         Class<?> type = parameter.getType();
         LOGGER.trace("test {}: Resolving parameter ({} {})", extensionContext.getUniqueId(), type.getSimpleName(), parameter.getName());
-        if (String.class.getTypeName().equals(type.getName())) {
-            if (parameter.getName().toLowerCase().contains("namespace")) {
-                return extractK8sNamespace(extensionContext);
-            }
+        if (String.class.getTypeName().equals(type.getName()) && parameter.getName().toLowerCase().contains("namespace")) {
+            return extractK8sNamespace(extensionContext);
         }
+
         return extensionContext;
     }
 
     @Override
     public void afterEach(ExtensionContext extensionContext) {
-        NamespaceUtils.deleteNamespaceWithWait(extractK8sNamespace(extensionContext));
+        String namespace = extractK8sNamespace(extensionContext);
+        Optional<Throwable> exception = extensionContext.getExecutionException();
+        if (exception.isPresent()) {
+            DeploymentUtils.collectClusterInfo(namespace, extensionContext.getRequiredTestClass().getSimpleName(), extensionContext.getRequiredTestMethod().getName());
+        }
+
+        NamespaceUtils.deleteNamespaceWithWait(namespace);
```

Comment:
```suggestion
        String namespace = extractK8sNamespace(extensionContext);
        try {
        Optional<Throwable> exception = extensionContext.getExecutionException();
        if (exception.isPresent()) {
            DeploymentUtils.collectClusterInfo(namespace, extensionContext.getRequiredTestClass().getSimpleName(), extensionContext.getRequiredTestMethod().getName());
        }
        } finally {

        NamespaceUtils.deleteNamespaceWithWait(namespace);
        }
```

Reply:
Why it is needed a `try` here?

Reply:
Got it, in case `collectClusterInfo` throws an exception we want to be sure we delete the namespace